CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+project(facil.io C)
+
+find_package(Threads REQUIRED)
+
+set(facil.io_SOURCES
```

Comment:
This is how you may find all C files automatically in CMake.

`file(GLOB_RECURSE facil.io_SOURCES LIST_DIRECTORIES false src/*.c)`

However it is not recommended by most CMake users as. Here is some explanation as to why:
https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil
Basically, using glob means that the user must run the command `cmake ..` to regenerate the build files after a `git pull` adds new C files. If the user doesn't realise that new files have been added they may be confused as to why a build fails to link.

I will modify the code if you do not mind the drawback.